proposals/fate-operator-proposal.md

```diff
@@ -0,0 +1,220 @@
+
+-   [Background](#background)
+-   [Motivation](#motivation)
+-   [Goals](#goals)
+-   [Non-Goals](#non-goals)
+-   [Design](#design)
+    -   [Container images and deploying FATE cluster on
+        Kubernetes](#container-images-and-deploying-fate-cluster-on-kubernetes)
+    -   [Custom Resource Definition](#custom-resource-definition)
+        -   [Kubefate](#kubefate)
+        -   [FATECluster](#fatecluster)
+        -   [FateJob](#fatejob)
+    -   [Controller design](#controller-design)
+-   [Reference](#reference)
+
+_Status_
+* 2020-5-28 - Draft v2
+* 2020-5-14 – Draft v1
+
+## Background
+Federated machine learning (FML) is a machine learning setting where many clients (e.g. mobile devices or organizations) collaboratively train a model under the coordination of a central server while keeping the training data decentralized. Only the encrypted mediate parameters are exchanged between clients with MPC or homomorphic encryption.
+![Federated Machine Learning](diagrams/fate-operator-fl.png)
+
+FML has received significant interest recently, because of its effectiveness to solve data silos and data privacy preserving problems. Companies participated in federated machine learning include 4Paradigm, ANT Financial, Data Republic, Google, Huawei, Intel, JD.com, Microsoft, Nvidia, OpenMind, Pingan Technology, Sharemind, Tencent, VMware, Webank etc. 
+
+Depending on the differences in features and sample data space, federated machine learning can be classified into horizontally federated machine learning, vertically federated machine learning and federated transfer learning. Horizontally federated machine learning also is called sample-based federated machine learning, means data sets share the same feature space but different in samples. With horizontally federated machine learning, we can gather the relatively small or partial data set into a big one to in-crease the performance of trained models. Vertical federated machine learning is applicable to the cases that two data set with different feature space but share same sample ID. With vertical federated machine learning we can train a model with attributes from different organizations for a full profile. Vertical federated machine learning is required to redesign most of machine learning algorithms. Federated transfer learning applies to scenarios that two data set with different features space but also different samples. 
```

Comment:
```suggestion
Depending on the differences in features and sample data space, federated machine learning can be classified into _horizontal federated machine learning_, _vertical federated machine learning_ and _federated transfer learning_. Horizontal federated machine learning is also called sample-based federated machine learning, which means data sets share the same feature space but have different samples. With horizontal federated machine learning, we can gather the relatively small or partial datasets into a big one to increase the performance of trained models. Vertical federated machine learning is applicable to the cases where there are two datasets with different feature space but share same sample ID. With vertical federated machine learning we can train a model with attributes from different organizations for a full profile. Vertical federated machine learning is required to redesign most of machine learning algorithms. Federated transfer learning applies to scenarios where there are two datasets with different features space but also different samples. 
```

Reply:
Thanks for the update suggestions, have merge in latest commit.